File: 7. Reverse Integer/reverse_number.cs
using System;

namespace rn
{
	class rn
	{
		static int reverse_number(int number, int radix = 10)
		{
			bool negative = (number < 0);
			
			number = (number < 0) ? (number * -1) : (number * 1);
			
			int nh = number, nje = 0, i = 1, mod;
			
			do{
				nje *= i;
				mod = nh % radix;
				nje += mod;
				nh -= mod;
				nh /= radix;
				i = 10;
			} while (nh > 0);
			
			if (negative){
				nje *= -1;
			}
			
			return (nje);
		}
		
		static void Main ()
		{
			int number = -1204;
			
			Console.WriteLine(reverse_number(number));
		}
	}
}

File: 7. Reverse Integer/reverse_number.rs
fn reverse_number(mut number: isize) -> isize{
	let base: isize = 10;
	let negative: bool = number < 0;
	
	number = if (number < 0) {(number * -1)} else {(number * 1)};
	
	let mut nh: isize = number;
	let mut out: isize = 0;
	let mut i: isize = 1;
	let mut modulo;
	
	while (nh > 0){
		out *= i;
		modulo = nh % base;
		out += modulo;
		nh -= modulo;
		nh /= base;
		i = 10;
	}
	
	if (negative){
		out *= -1;
	}
	
	return (out);
}

fn main (){
	println!("{}", reverse_number(-35));
}
